gtfu/src/main/java/gtfu/tools/TripListGenerator.java

```diff
@@ -177,6 +192,23 @@ private static Map<String, String> createCustomDirectionMap() {
         return directionMap;
     }
 
+    private static Map<String, List<String>> createServiceIDMap(int exception, CalendarDateCollection calendarDates){
+        Map<String, List<String>> dateMap = new HashMap();
+        for (CalendarDate cd : calendarDates){
+            if(cd.getExceptionType() == exception){
+                List<String> dateList = dateMap.get(cd.getServiceID());
```

Comment:
Instead of the below, it should be:
```
List<String> dateList = dateMap.get(cd.getServiceID());

if (dateList == null){
    dateList = new ArrayList();
    dateMap.put(cd.getServiceID(), dateList);
}

dateList.add(cd.getDate());
```